common/djangoapps/third_party_auth/signals/tests/test_handlers.py

```diff
@@ -115,3 +117,205 @@ def test_saml_config_signal_handlers(
+        {
+            'test_case': 'cross_site',
+            'description': 'cross-site provider updated',
+            'setup_data': {
+                'create_valid_provider': False,
+                'create_cross_site_provider': True,
+                'create_site2_provider': False,
+            },
+        },
+    )
+    @ddt.unpack
+    @override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=True)
+    def test_saml_provider_config_updates(self, test_case, description, setup_data):
+        """
+        Test SAML provider config updates under different scenarios.
+        """
+        if test_case == 'null_preserved':
+            valid_provider_config = SAMLProviderConfigFactory(
+                slug='valid_provider',
+                site=self.site1,
+                saml_configuration=self.config1
+            )
+
+            new_config = self._create_new_config(self.site1)
+
+            current_valid = self._get_current_provider('valid_provider')
+            current_null = self._get_current_provider('null_provider')
+
+            # Verify: valid provider updated, NULL provider unchanged
+            self.assertEqual(current_valid.saml_configuration_id, new_config.id)
+            self.assertIsNone(current_null.saml_configuration_id)
+
+        elif test_case == 'site_isolation':
+            site1_provider_config = SAMLProviderConfigFactory(
+                slug='site1_provider',
+                site=self.site1,
+                saml_configuration=self.config1
+            )
+            site2_provider_config = SAMLProviderConfigFactory(
+                slug='site2_provider',
+                site=self.site2,
+                saml_configuration=self.config2
+            )
+            original_site2_config_id = site2_provider_config.saml_configuration_id
+
+            new_config = self._create_new_config(self.site1)
+
+            current_site1 = self._get_current_provider('site1_provider')
+            current_site2 = self._get_current_provider('site2_provider')
+
+            # Verify: only site1 provider updated, site2 provider unchanged
+            self.assertEqual(current_site1.saml_configuration_id, new_config.id)
+            self.assertEqual(current_site2.saml_configuration_id, original_site2_config_id)
+
+        elif test_case == 'cross_site':
+            cross_site_provider_config = SAMLProviderConfigFactory(
+                slug='cross_site_provider',
+                site=self.site2,
+                saml_configuration=self.config1
+            )
+
+            new_config = self._create_new_config(self.site1)
+
+            # Provider should be updated because its SAML config is from site1
+            current_provider = self._get_current_provider('cross_site_provider')
+            self.assertEqual(current_provider.saml_configuration_id, new_config.id)
+
+    @override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=True)
+    def test_null_configuration_preserved(self):
```

Comment:
I believe this and all the following tests are redundant to the ddt test and can be removed? Maybe that wasn't the case with the original ddt case, but I was struggling to understand what cases it covered.